features/conversation/conversation-messages-list.tsx

```diff
@@ -21,37 +29,105 @@ export const ConversationMessagesList = memo(function ConversationMessagesList(
+          index: messageIds.indexOf(scrollToMessageId),
+          animated: true,
+          viewOffset: 100, // Random value just so that the message is not directly at the bottom
+        });
+        conversationStore.setState({
+          scrollToMessageId: undefined,
+        });
+      }
+    );
+
+    return () => {
+      unsub();
+    };
+  }, [conversationStore, messageIds, scrollRef]);
+
+  // WIP. Lib isn't ready yet
+  // return (
+  //   <LegendList
+  //     data={messageIds}
+  //     renderItem={({ item, index }) =>
+  //       renderMessage({
+  //         messageId: item,
+  //         index,
+  //       })
+  //     }
+  //     itemLayoutAnimation={theme.animation.reanimatedLayoutSpringTransition}
+  //     keyboardDismissMode="interactive"
+  //     estimatedItemSize={100}
+  //     maintainScrollAtEnd
+  //     alignItemsAtEnd
+  //     automaticallyAdjustContentInsets={false}
+  //     contentInsetAdjustmentBehavior="never"
+  //     keyExtractor={keyExtractor}
+  //     keyboardShouldPersistTaps="handled"
+  //     ItemSeparatorComponent={MessageSeparator}
+  //     showsVerticalScrollIndicator={Platform.OS === "ios"} // Size glitch on Android
+  //     pointerEvents="auto"
+  //     /**
+  //      * Causes a glitch on Android, no sure we need it for now
+  //      */
+  //     // maintainVisibleContentPosition={{
+  //     //   minIndexForVisible: 0,
+  //     //   autoscrollToTopThreshold: 100,
+  //     // }}
+  //     // estimatedListSize={Dimensions.get("screen")}
+  //     {...rest}
+  //   />
+  // );
+
   return (
-    // @ts-ignore
-    <Animated.FlatList
-      inverted
+    <AnimatedFlashList
+      inverted={true}
+      ref={scrollRef}
       data={messageIds}
       renderItem={({ item, index }) =>
         renderMessage({
           messageId: item,
           index,
         })
       }
-      itemLayoutAnimation={theme.animation.reanimatedLayoutSpringTransition}
+      // itemLayoutAnimation={theme.animation.reanimatedLayoutSpringTransition}
       keyboardDismissMode="interactive"
-      automaticallyAdjustContentInsets={false}
-      contentInsetAdjustmentBehavior="never"
+      // automaticallyAdjustContentInsets={false}
+      // contentInsetAdjustmentBehavior="never"
       keyExtractor={keyExtractor}
       keyboardShouldPersistTaps="handled"
-      ItemSeparatorComponent={() => <MessageSpaceBetweenMessages />}
+      ItemSeparatorComponent={() => <MessageSeparator />}
       showsVerticalScrollIndicator={Platform.OS === "ios"} // Size glitch on Android
-      pointerEvents="auto"
+      // pointerEvents="auto"
       /**
        * Causes a glitch on Android, no sure we need it for now
        */
       // maintainVisibleContentPosition={{
       //   minIndexForVisible: 0,
       //   autoscrollToTopThreshold: 100,
       // }}
-      // estimatedListSize={Dimensions.get("screen")}
+      estimatedItemSize={50} // 50 is very random for now
+      estimatedListSize={{
```

Comment:
is estimatedListSize the size of the entire list or just the amount of space it will be shown on screen?

Reply:
The size it will show on screen. It's optional but helps with performance

Reply:
thnx til